helm/v1beta1/types_project_helm_chart_repository.go

```diff
@@ -19,13 +19,60 @@ type ProjectHelmChartRepository struct {
 	// spec holds user settable values for configuration
 	// +kubebuilder:validation:Required
 	// +required
-	Spec HelmChartRepositorySpec `json:"spec"`
+	Spec ProjectHelmChartRepositorySpec `json:"spec"`
 
 	// Observed status of the repository within the namespace..
 	// +optional
 	Status HelmChartRepositoryStatus `json:"status"`
 }
 
+// Project Helm chart repository exposed within a namespace
+type ProjectHelmChartRepositorySpec struct {
+
+	// If set to true, disable the repo usage in the cluster/namespace
+	// +optional
+	Disabled bool `json:"disabled,omitempty"`
```

Comment:
note that this bool must remain because this is an existing serialization.  In retrospect, this should have something like ManagementState we use for operators.

Reply:
Are you suggesting to change the type for Disabled type from bool to ManagementState?

Reply:
> Are you suggesting to change the type for Disabled type from bool to ManagementState?

No, we cannot change the type. I only meant to acknowledge the debt.